### DISCUSSION TITLE
CGAL Formula for Centroid in 2D polygon

### DISCUSSION THREAD

**Discussion by Alison-97** [CGAL/cgal#7131] [General]
I'm using CGAL 5.5 and I tried to compute centroid for a simple 2D polygon. When using the formula for composite shapes the resulting centroid is **(2.63, 1.7)** as shown below. [[Source](https://www.geogebra.org/m/nbexg5v9)]
![image](https://user-images.githubusercontent.com/97237245/208820694-6274fa6f-d38d-474b-b5ba-bf991e727909.png)
But when I'm using CGAL::centroid, the centroid returns **(2.5, 1.5)**. What could be causing this discrepancy?

```
int main()
{
  // centroid of 2D points
  std::vector<Point_2> points_2;
  points_2.push_back(Point_2(0, 0));
  points_2.push_back(Point_2(3, 0));
  points_2.push_back(Point_2(7, 3));
  points_2.push_back(Point_2(0, 3));
  Point_2 c2 = CGAL::centroid(points_2.begin(), points_2.end(),CGAL::Dimension_tag<0>());
  std::cout  << c2 << std::endl;		//  2.5, 1.5

  return 0;
}
```

**Comment by soesau** [comment]
Hi,

The function CGAL::centroid takes as a last input the dimension of the input, i.e., CGAL::Dimension_tag<0>(). In your case, you provide the function iterators of points and the dimension 0. If instead you provide triangles and CGAL::Dimension_tag<2>(), you will get the centroid of the 2d polygon. There is also the [centroid example](https://doc.cgal.org/latest/Principal_component_analysis/Principal_component_analysis_2centroid_8cpp-example.html) in the Principal component analysis package, that shows the difference.

**Comment by afabri** [comment]
@MaelRL  @palliez when computing the area of a polygon on adds the signed areas of triangles formed by the origin and two consecutive polygon vertices. Can something similar be done for the centroid of a polygon, that is without triangulating the polygon explicitly?

**Reply by MaelRL** [reply]
If the polygon has a non-empty kernel, you can do something similar with a point in the kernel, but otherwise no.
